src/NuGet.Clients/NuGet.PackageManagement.UI/UserInterfaceService/NuGetProjectUpgradeHelper.cs

```diff
@@ -20,79 +20,9 @@ namespace NuGet.PackageManagement.UI
 {
     public static class NuGetProjectUpgradeHelper
     {
-        private static readonly HashSet<string> UpgradeableProjectTypes =
-            new HashSet<string>(StringComparer.OrdinalIgnoreCase)
-            {
-                VsProjectTypes.CsharpProjectTypeGuid,
-                VsProjectTypes.VbProjectTypeGuid,
-                VsProjectTypes.FsharpProjectTypeGuid
-            };
-
-        private static readonly HashSet<string> UnupgradeableProjectTypes =
-            new HashSet<string>(StringComparer.OrdinalIgnoreCase)
-            {
-                VsProjectTypes.CppProjectTypeGuid,
-                VsProjectTypes.WebApplicationProjectTypeGuid,
-                VsProjectTypes.WebSiteProjectTypeGuid
-            };
-
         public static async Task<bool> IsNuGetProjectUpgradeableAsync(NuGetProject nuGetProject, Project envDTEProject = null)
```

Comment:
why even have this method then? Every caller of this method should call NuGetProjectUpgradeUtility instead

Reply:
Because this code is already part of 15.7 preview bits and I don't want to change the interface at this time to avoid any patch related issue afterwards.

Reply:
those issues happen when these changes are spread across DLLs that are used by both CLI and VS. This is a VS only DLL

Reply:
it can happen in VS as well, besides we shouldn't have breaking changes in patch releases or should we?